atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/SqliteOracleAdapter.java

```diff
@@ -0,0 +1,309 @@
+            PreparedStatement statement = connection.prepareStatement("SELECT table_name FROM " + METADATA_TABLE);
+            ResultSet resultSet = statement.executeQuery();
+
+            ImmutableSet.Builder<TableReference> tableNames = ImmutableSet.builder();
+            while (resultSet.next()) {
+                tableNames.add(TableReference.createFromFullyQualifiedName(resultSet.getString("table_name")));
+            }
+            return tableNames.build();
+        });
+    }
+
+    public Set<String> listAllTablesWithMapping() {
+        return runWithConnection(connection -> {
+            PreparedStatement statement =
+                    connection.prepareStatement("SELECT table_name FROM " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE);
+            ResultSet resultSet = statement.executeQuery();
+
+            ImmutableSet.Builder<String> tableNames = ImmutableSet.builder();
+            while (resultSet.next()) {
+                tableNames.add(resultSet.getString("table_name"));
+            }
+            return tableNames.build();
+        });
+    }
+
+    private Connection applyPragma(Connection connection) throws SQLException {
+        // Oracle likes are case-sensitive. This pragma ensures that we mimic that behaviour with our sqlite queries
+        connection.createStatement().executeUpdate("PRAGMA case_sensitive_like = true;");
+        return connection;
+    }
+
+    private String getConnectionUrl() {
+        return "jdbc:sqlite:" + sqliteFileLocation.getAbsolutePath();
+    }
+
+    private String generatePrefixedTableName(String prefix, TableReference tableReference) {
+        OracleDdlConfig config = ImmutableOracleDdlConfig.builder()
+                .tablePrefix(prefix)
+                .overflowMigrationState(OverflowMigrationState.UNSTARTED)
+                .build();
+
+        OracleTableNameGetter tableNameGetter = OracleTableNameGetter.createDefault(config);
+        return tableNameGetter.getPrefixedTableName(tableReference);
+    }
+
+    /**
+     * A lightweight wrapper around {@link SimpleTimedSqlConnectionSupplier} that marshals some queries containing
+     * Oracle specific keywords such as PURGE to sqlite compatible queries
+     */
+    private static class SqliteSqlConnectionSupplier implements SqlConnectionSupplier {
+        private final SqlConnectionSupplier delegate;
+
+        SqliteSqlConnectionSupplier(ConnectionSupplier delegate) {
+            this.delegate = new SimpleTimedSqlConnectionSupplier(delegate);
+        }
+
+        @Override
+        public SqlConnection get() {
+            return marshalOracleQueryToSqlite(delegate.get());
+        }
+
+        @Override
+        public void close() throws PalantirSqlException {
+            delegate.close();
+        }
+
+        private SqlConnection marshalOracleQueryToSqlite(SqlConnection sqlConnection) {
+            SqlConnection spy = spy(sqlConnection);
+            doAnswer(invocation -> {
```

Comment:
This is clever, but also feels fragile - it seems to rely on "if statement ends in PURGE then it must be a DROP TABLE" statement, which I don't know is the case. Could we also check that the command is a `DROP TABLE` command of some kind?

Reply:
There's DROP TABLE and PURGE commands, both of which contain the word purge (but only the former actually has PURGE as a keyword at the end).

This doesn't actually rely on that it's a drop table command (more that sqlite doesn't have a PURGE keyword on any statement), since it just strips purge from the end. I've made it less fragile though by checking it starts with DROP TABLE.